core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -66,6 +68,8 @@ public void close() {
   private class CallCredentialsApplyingTransport extends ForwardingConnectionClientTransport {
     private final ConnectionClientTransport delegate;
     private final String authority;
+    private final AtomicLong pendingApplier = new AtomicLong(0);
+    private volatile CallCredentialsApplyingTransportListener listener;
```

Comment:
non-volatile. Start is guaranteed to be called before newStream.

Reply:
Race between `start()` and `onComplete()`? metadataApplier referring to this listener `onComplete` concurrently.

Reply:
onComplete() is called after newStream() which is called after start(). So start() is guaranteed to be called before onComplete().